### DISCUSSION TITLE
How to enable HTTPS in dev?

### DISCUSSION THREAD

**Discussion by flowreaction** [nuxt/nuxt#15891] [Questions] [answered]
Using the current RC9 version, I tried getting HTTPS to work for me on my Mac, but can't seem to get it right.
I used mkcert to create a localhost.pem and localhost-key.pem, and a set the dev script in my package.json to this:
`nuxt dev --https  --ssl-cert localhost.pem --ssl-key localhost-key.pem`
I also have a .env in my project root with the following entry: `NODE_TLS_REJECT_UNAUTHORIZED=0`. This should work according to the documentation.
I get the following error:

```
request to http://0.0.0.0:3000/__nuxt_vite_node__/manifest failed, reason: socket hang up ()

at async $fetchRaw2 (./node_modules/ohmyfetch/dist/chunks/fetch.mjs:131:20)
at async ./.nuxt/dev/index.mjs:457:20
at async ./.nuxt/dev/index.mjs:520:64
at async ./.nuxt/dev/index.mjs:106:22
at async ./node_modules/h3/dist/index.mjs:592:19
at async Server.nodeHandler (./node_modules/h3/dist/index.mjs:538:7)
```

So I am not sure if I have configured something wrong or if this is a bug. I would greatly appreciate any help.

**Comment by danielroe** [comment, marked as answer]
This should be resolved on the [edge channel](https://v3.nuxtjs.org/guide/going-further/edge-channel/) in https://github.com/nuxt/framework/pull/7271, though you will need to enable `NODE_TLS_REJECT_UNAUTHORIZED=0`? https://v3.nuxtjs.org/api/commands/dev/#nuxi-dev.

**Reply by toniengelhardt** [reply]
If I follow the instructions of OP I get at `https://localhost:3000`

<img width="715" alt="Screenshot 2023-01-18 at 11 34 25" src="https://user-images.githubusercontent.com/2703233/213162594-1766bd25-a300-49bb-b062-bdf535a24d52.png">

and then when I proceed the Nuxt default screen

<img width="1010" alt="Screenshot 2023-01-18 at 11 39 16" src="https://user-images.githubusercontent.com/2703233/213162716-e3920fdb-f11e-48a4-8423-b91c564add0c.png">

**Reply by DeltaCorporate** [reply]
try advanced and then continue

**Reply by toniengelhardt** [reply]
I did, but then I just get the "Welcome" screen instead of my app 😔

**Reply by DeltaCorporate** [reply]
ok so this is due to code issue. Please follow these steps:

delete app.vue

create pages/index.vue with home page html

restart app

That should resolve your problem

**Reply by toniengelhardt** [reply]
I don't think it's a code issue, the app works perfectly fine without the `--https` flag, you can see it here: https://github.com/toniengelhardt/merklin

**Reply by DeltaCorporate** [reply]
where are your certificates?

**Reply by toniengelhardt** [reply]
In the project root, like specified here: https://www.storyblok.com/faq/setting-up-https-on-localhost-in-nuxt-3

**Reply by DeltaCorporate** [reply]
and what about NODE_TLS_REJECT_UNAUTHORIZED=0? and whut about the command?

**Reply by toniengelhardt** [reply]
I added `NODE_TLS_REJECT_UNAUTHORIZED=0` to my `.env` file and launching with 

```bash
nuxt dev --https  --ssl-cert localhost.pem --ssl-key localhost-key.pem
```

Might there be a problem with the certificate itself? But the question remains why I get the default screen...

**Reply by DeltaCorporate** [reply]
that could be a problem with the security level of the browser

**Reply by toniengelhardt** [reply]
I'm on latest Chrome. How to check?

**Reply by DeltaCorporate** [reply]
try on brave or firefox browsers

**Reply by toniengelhardt** [reply]
Same behavior on Brave and Safari. Firefox doesn't let me open the page at all.

<img width="819" alt="Screenshot 2023-01-19 at 11 18 12" src="https://user-images.githubusercontent.com/2703233/213429485-f2077c31-58e5-4dc6-90ce-d94fbf801e6f.png">

What's also strange is that if I start the server with 

```bash
nuxt dev --https --ssl-cert localhost.pem ssl-key localhost-key.pem
```

Nuxt ignores my `devServer` settings for `host` and `port` and always launches on `3000`. Without the `--https` flag it works. But I also tried to remove the `devServer` settings and it doesn't change anything.

```ts
devServer: {
    host: process.env.HOST || 'localhost',
    port: process.env.PORT ? +process.env.PORT : 3000,
},
```

How do you generate your certs @DeltaCorporate, with mkcert?

**Reply by userquin** [reply]
for FF you can add the url to FF exceptions, check this comment: https://github.com/elk-zone/elk/pull/337#issuecomment-1339743248

**Reply by toniengelhardt** [reply]
Thanks @userquin! Ideally I would want to test an installed PWA in Chrome on my local WiFi network.

**Reply by userquin** [reply]
I use `https-localhost`, it will create a TLS certificate (the certificate is also self signed CA) when you run `serve` first time and you can reuse it across multiple projects. 

`https-localhost` will also ask you to install the certificate in OS (keychain in macOS and certificate manager in Windows: not tested in linux).

You can check `elk` repo, I added a few scripts to run https on dev (check `https` script and this https://github.com/elk-zone/elk/tree/main/https-dev-config folder: the TLS certificate used in the repo is my local copy generated by `https-localhost`.

You can check how to use `https-localhost` in `vite-plugin-pwa` repo, just add the dev dependency and run `node/express`: you can install the certificate then you can reuse it from another projects, beware, adding it to a nuxt 3 project, I don't know why it is not working.

**Reply by userquin** [reply]
If you want to test on local wifi other than localhost you will need to use another tool to generate the TLS certificate: maybe https://github.com/vitejs/vite-plugin-basic-ssl, check this PR https://github.com/vitejs/vite-plugin-basic-ssl/pull/13 (some hints for installation https://github.com/vitejs/vite-plugin-basic-ssl/blob/8d3828b81b0935060c32aae91bd798f56e76cda6/README.md)

**Reply by toniengelhardt** [reply]
Amazing! Thanks a lot, I'll give it a try.

**Comment by DeltaCorporate** [comment]
Just to add a small clarification, you can create an .env file in which you put NODE_TLS_REJECT_UNAUTHORIZED=0.
